videoprocess/vpphdr_tm.cpp

```diff
@@ -0,0 +1,861 @@
+*
+* The above copyright notice and this permission notice shall be included
+* in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+* OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+* THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+* OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+* ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+* OTHER DEALINGS IN THE SOFTWARE.
+*/
+/*
+ * Video process test case based on LibVA.
+ * This test covers high dynamic range tone mapping feature.
+ * Usage: ./vpphdr_tm process_hdr_tm.cfg
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <assert.h>
+#include <va/va.h>
+#include <va/va_vpp.h>
+#include "va_display.h"
+
+#ifndef VA_FOURCC_I420
+#define VA_FOURCC_I420 0x30323449
+#endif
+
+#define MAX_LEN   1024
+
+#define CHECK_VASTATUS(va_status,func)                                      \
+  if (va_status != VA_STATUS_SUCCESS) {                                     \
+      printf("%s:%s (%d) failed,exit\n", __func__, func, __LINE__);         \
+      exit(1);                                                              \
+  }
+
+static VADisplay va_dpy = NULL;
+static VAContextID context_id = 0;
+static VAConfigID  config_id = 0;
+static VASurfaceID g_in_surface_id = VA_INVALID_ID;
+static VASurfaceID g_out_surface_id = VA_INVALID_ID;
+
+static FILE* g_config_file_fd = NULL;
+static FILE* g_src_file_fd = NULL;
+static FILE* g_dst_file_fd = NULL;
+
+static char g_config_file_name[MAX_LEN];
+static char g_src_file_name[MAX_LEN];
+static char g_dst_file_name[MAX_LEN];
+
+static uint32_t g_in_pic_width = 1920;
+static uint32_t g_in_pic_height = 1080;
+static uint32_t g_out_pic_width = 1920;
+static uint32_t g_out_pic_height = 1080;
+
+static uint32_t g_in_fourcc  = VA_FOURCC('N', 'V', '1', '2');
+static uint32_t g_in_format  = VA_FOURCC_P010;
+static uint32_t g_out_fourcc = VA_FOURCC('N', 'V', '1', '2');
+static uint32_t g_out_format = VA_RT_FORMAT_YUV420;
+static uint32_t g_src_file_fourcc = VA_FOURCC('I', '4', '2', '0');
+static uint32_t g_dst_file_fourcc = VA_FOURCC('Y', 'V', '1', '2');
+
+static uint32_t g_frame_count = 1;
+static uint32_t g_in_max_display_luminance = 1000;
+static uint32_t g_in_min_display_luminance = 1;
+static uint32_t g_in_max_content_luminance = 4000;
+static uint32_t g_in_pic_average_luminance = 1000;
+static uint32_t g_out_max_display_luminance = 1000;
+static uint32_t g_out_min_display_luminance = 1;
+static uint32_t g_out_max_content_luminance = 4000;
+static uint32_t g_out_pic_average_luminance = 1000;
+
+static uint32_t g_in_colour_primaries = 9;
+static uint32_t g_in_transfer_characteristic = 16;
+
+static uint32_t g_out_colour_primaries = 9;
```

Comment:
could we use macro or enum for these const?

Reply:
In VAAPI, https://github.com/intel/libva/blob/master/va/va_vpp.h#L715, colour_primaries is defined as uint8_t. Hence, here, I used some integer. 

Please let me know your feedback on this.

Reply:
If you think it is difficult to read, maybe need to think about a way to improve that. Please let me know your feedback. Thank you!

Reply:
from the your comments: https://github.com/intel/libva/commit/7bbddffddb4a15f0163f89d5d87860866e850cde
the value of primaries is :1, 6, 9 and characteristic is 1, 4, 6,8, 13, 14,15, 16.  looks hard to using a simple string to describe it